analyzers/tests/SonarAnalyzer.UnitTest/Rules/ArrayPassedAsParamsTest.cs

```diff
@@ -44,9 +44,13 @@ public class ArrayPassedAsParamsTest
     [TestMethod]
     public void ArrayPassedAsParams_CSharp12() =>
         builderCS.AddSnippet("""
-            MyClass s = new(1, [1, 2, 3]); // FN
+            MyClass s0 = new(1, [1, 2, 3]);     // Noncompliant
+            MyClass s1 = new(1, []);            // Noncompliant
+            MyClass2 s2 = new([1], [1, 2, 3]); // Noncompliant
+            MyClass2 s3 = new([1, 2, 3], 1);
 
             public class MyClass(int a, params int[] args) { }
+            public class MyClass2(int[] a, params int[] args) { }
             """)
```

Comment:
Can we test with multidimensional arrays and multidimensional collection expressions?
```cs
MyClass3 s3 = new([1, 2, 3], [4, 5, 6]); // Compliant
MyClass3 s4 = new([[1, 2, 3], [4, 5, 6]]); // Noncompliant

public class MyClass3(params int[][] args);
```

Personal opinion 1: I would remove the `public` on the class, as well as its empty body since they are not relevant to the test. 

Personal opinion 2: If you use the discard you don't have to maintain ordinals in variable names: `_ = new MyClass3(...)`, whenever you add a new case.

Reply:
Sure, I added those. FPs on jugged arrays are already known and documented, if I have time I might fix those as well during this sprint (not strictly C#12 related, but frequently requested by the community, see #8163 and #6894)